src/converter.js

```diff
@@ -58,27 +58,25 @@ class Converter {
     return `Array<${this.toUnion(types)}>`;
   }
   toLiteralUnion(values) {
-    const types = values.map(v => (typeof v === 'string' ? `'${v}'` : v));
+    const types = values.map((v) => (typeof v === 'string' ? `'${v}'` : v));
     return this.toUnion(types);
   }
   toJSDoc(str) {
-    const docs = str
-      .split('\n')
-      .join('\n * ')
-      .replace(/\*\//g, '*\\/');
+    const docs = str.split('\n').join('\n * ').replace(/\*\//g, '*\\/');
     return `\n/**\n * ${docs}\n */\n`;
   }
   toObject(props = {}, additional = false) {
     const fields = [];
 
     Object.keys(props)
       .sort()
-      .forEach(name => {
+      .forEach((name) => {
+        const propName = name.includes('.') ? `'${name}'` : name;
```

Comment:
Which API does this?

Reply:
it's from a private api from my employer. The API includes a few proto field in GET request parameters thus the generated discovery json flattened it to allow them to be part of query string. 

Imaging `user.name` and `user.role` from an User message.

Reply:
Please let me know if you have any concern, thank you!

Reply:
Update: This will also solve the error when running `dtsd bigquery v2 > bigquery.d.ts`

You can find in [bigquery v2](https://bigquery.googleapis.com/$discovery/rest?version=v2)
```
"formatOptions.useInt64Timestamp": {
  "description": "Optional. Output timestamp as usec int64. Default is false.",
  "location": "query",
  "type": "boolean"
}
```